File: API/Entities/MealPlan.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.AspNetCore.Components;

namespace API.Entities;

public class MealPlan
{
    public int Id { get; set; }      
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public Photo? Photo { get; set; } = null!;  
    public ICollection<MealPlanRecipe> MealPlanRecipes { get; set; } = [];
    public int UserId { get; set; }
    public DateTime StartDate { get; set; }
public DateTime EndDate { get; set; }

    public ApplicationUser User { get; set; } = null!;
}

File: client/src/app/_models/mealPlan.ts
// Updated _models/mealPlan.ts
import { MealPlanRecipe } from "./mealPlanRecipe";
import { Photo } from "./photo";

export interface MealPlan {
  id: number;
  name: string;
  description: string;
  photo?: Photo;
  startDate: Date | string;
  endDate: Date | string;
  userId: number;
  mealPlanRecipes: MealPlanRecipe[];
}